a3fe/run/stage.py

```diff
@@ -175,6 +175,10 @@ def __init__(
                 self.lam_vals = lambda_values
             else:
                 self.lam_vals = self._get_lam_vals()
+
+            if self.engine_config is not None:
+                self.engine_config.lambda_values = self.lam_vals
+
```

Comment:
Please remove all of this logic. Do not allow the user to pass a set of lambda values - this violates the single source of information principle because then the values are also in the config. Please just create a property which gets the lambda values from the engine config.